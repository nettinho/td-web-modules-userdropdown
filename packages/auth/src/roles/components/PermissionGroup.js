import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { compose } from "redux";
import { Header, List, Checkbox, Segment } from "semantic-ui-react";
import { injectIntl } from "react-intl";
import { updateRolePermissions } from "../routines";

export const PermissionGroup = ({
  name,
  permissions,
  rolePermissions,
  roleId,
  updateRolePermissions,
  intl: { formatMessage }
}) => (
  <Segment>
    <Header as="h4" content={name} />
    <List size="small">
      {permissions.map((p, i) => (
        <List.Item key={i}>
          <Checkbox
            label={{ children: formatMessage({ id: `permission.${p.name}` }) }}
            checked={_.find(_.propEq("id", p.id))(rolePermissions) && true}
            onChange={(e, data) => {
              const nextRolePermissions = data.checked
                ? _.concat(p)(rolePermissions)
                : _.filter(_.negate(_.propEq("id", p.id)))(rolePermissions);
              updateRolePermissions({
                id: roleId,
                permissions: nextRolePermissions
              });
              e.preventDefault();
            }}
          />
        </List.Item>
      ))}
    </List>
  </Segment>
);

PermissionGroup.propTypes = {
  roleId: PropTypes.number,
  intl: PropTypes.object,
  name: PropTypes.string,
  permissions: PropTypes.array,
  rolePermissions: PropTypes.array,
  updateRolePermissions: PropTypes.func
};

const mapStateToProps = ({ role, rolePermissions }) => ({
  roleId: _.get("id")(role),
  rolePermissions
});

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ updateRolePermissions }, dispatch)
});

export default compose(
  injectIntl,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(PermissionGroup);
