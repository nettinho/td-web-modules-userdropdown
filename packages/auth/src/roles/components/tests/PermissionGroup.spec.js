import React from "react";
import { shallowWithIntl } from "intl-stub";
import { PermissionGroup } from "../PermissionGroup";

describe("<PermissionGroup />", () => {
  const id = 1;
  const name = "Taxonomy";
  const permissions = [{ id: 1, name: "perm1" }, { id: 4, name: "perm4" }];
  const rolePermissions = [{ id: 1, name: "perm1" }, { id: 3, name: "perm3" }];
  const props = { id, name, rolePermissions, permissions };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<PermissionGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
