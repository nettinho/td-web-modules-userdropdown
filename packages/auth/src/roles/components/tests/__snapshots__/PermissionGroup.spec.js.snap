// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PermissionGroup /> matches the latest snapshot 1`] = `
<Segment>
  <Header
    as="h4"
    content="Taxonomy"
  />
  <List
    size="small"
  >
    <ListItem
      key="0"
    >
      <Checkbox
        checked={true}
        label={
          Object {
            "children": "permission.perm1",
          }
        }
        onChange={[Function]}
        type="checkbox"
      />
    </ListItem>
    <ListItem
      key="1"
    >
      <Checkbox
        label={
          Object {
            "children": "permission.perm4",
          }
        }
        onChange={[Function]}
        type="checkbox"
      />
    </ListItem>
  </List>
</Segment>
`;
